test/Common/SupportedLanguagesTestAttribute.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using Xunit.Sdk;
+using System.Reflection;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql.Tests.Common
+{
+    /// <summary>
+    /// Base attribute class for adding language parameter data to tests. This allows
+    /// tests to be run against all the supported languages list that are specified in the SupportedLanguages
+    /// list of this class (ex: CSharp, JavaScript)
+    /// </summary>
+    public abstract class LanguageTestAttribute : DataAttribute
+    {
+        // add all the list of supported languages.
+        protected enum SupportedLanguages
+        {
+            CSharp
+        }
+        protected readonly Dictionary<string, List<List<object>>> testData = new Dictionary<string, List<List<object>>>();
+        protected readonly string _propertyName = "data";
+
+        /// <inheritDoc />
+        public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+        {
+            if (testMethod == null) { throw new ArgumentNullException(nameof(testMethod)); }
+
+            if (this.testData.Count == 0) { throw new EmptyException(nameof(testMethod)); }
+            string serializedData = JsonConvert.SerializeObject(this.testData);
+            var allData = JObject.Parse(serializedData);
+            JToken data = allData[this._propertyName];
+            return data.ToObject<List<object[]>>();
+        }
+
+    }
+    public class SupportedLanguagesAttribute : LanguageTestAttribute
```

Comment:
As we discussed, we only want one attribute. By default all tests should be running against all languages, and then we should have an `UnsupportedLanguages` attribute that tells it when there are languages not supported for a test currently. 

Looking at this implementation I think we should refactor this slightly - have a single attribute 

`public class SqlInlineData extends DataAttribute`

The languages enum would be a [Flag](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-6.0) enumeration (and I'd suggest renaming it to just something like `TargetLanguage`)

Then there'd be two constructors for SqlInlineData : 

`public SqlInlineData(TargetLanguage unsupportedLanguages, params object[] args)`
`public SqlInlineData(params object[] args)`

Tests that supported all languages would use the second, ones that wanted to exclude languages would use the first (using the flag syntax, e.g. `TargetLanguage.Python | TargetLanguage.Javascript`)

Examples : 

`[SqlInlineData()]` - supports all languages, no params
`[SqlInlineData(3, "foo")]` - supports all languages, has params
`[SqlInlineData(TargetLanguage.Python)]` - doesn't support Python, no params
`[SqlInlineData(TargetLanguage.Python, 3, "foo")]` - doesn't support Python, has params

Reply:
Actually, looking at DataAttribute it might be better to make two attributes, `SqlInlineData` and `UnsupportedLanguages`. SqlInlineData would extend DataAttribute and then use the MethodInfo passed in to `GetData` to look for `UnsupportedLanguages` attribute on that method - and if so read it to get the list of unsupported languages. 

So it'd look like this on the test method

```
[SqlInlineData(3, "foo")]
[UnsupportedLanguages(TargetLanguage.Python, TargetLanguage.Javascript)]
```

Reply:
So simply have one attribute class and remove the rest. Sounds good to me. But can I suggest a change, instead of having the list of unsupported ones, what if I make the logic for support-specific-language(s). 

Examples 

[SqlInlineData(TargetLanguage.Python)] - supports only Python, no params
[SqlInlineData(TargetLanguage.Python, 3, "foo")] - supports only Python, has params

Usually it's more like a test is specific for a language and is not applicable to others so having it the above way reflects that.

Reply:
I'd rather not, we should expect that most of the tests support all languages (that should be the goal when writing tests). So it should be rarer to need to specify unsupported languages. 

As long as it's understood that tests by default using the SqlInlineData run against all languages that should be fine.

Reply:
And to clarify - I think it'd be best to have two attributes. `SqlInlineData` (which is only used to specify the parameters if they exist) and `UnsupportedLanguages` (which only exists if the test doesn't support some languages)

Reply:
For some reason my browser didn't refresh your second suggestion of having two attributes (reading the UnsupportedLanguages attribute from method info part). Ok, I still think the support-specific-languages will work better and is a simpler implementation but having it as you suggested also works with a little bit more work. I'll get working on it. Thanks!